pkg/os/util_linux.go

```diff
@@ -18,6 +20,10 @@ func WriteToFileAsRoot(reason, content, filepath string) error {
 	if err := cmd.Run(); err != nil {
 		return fmt.Errorf("Failed writing to file as root: %s: %s: %v", filepath, buf.String(), err)
 	}
+	if _, _, err := RunWithPrivilege(fmt.Sprintf("Changing permission for %s to %d ", filepath, mode),
+		"chmod", strconv.FormatUint(uint64(mode), 8), filepath); err != nil {
```

Comment:
Could also be `fmt.Sprintf("%o", uint64(mode))`

Reply:
os.FileMode is a `uint32`, so maybe slightly more expected to cast to `uint32`?

Reply:
@cfergeau yes but `strconv.FormatUint` doesn't allow to use uint32 so have to cast. Also as per https://stackoverflow.com/questions/24886015/how-to-convert-uint32-to-string using `strconv.FormatUint` is much better than `fmt.Sprintf`

Reply:
> The better option is strconv.FormatUint because it is faster, has less memory allocations (benchmark examples here or here).

This is not really relevant in this code as we will not be doing the conversion really often, and this is used in a function which called `system()`.